### ISSUE TITLE
Multiple Handlers?

### ISSUE THREAD

**Issue by sstream17** [evanlindsey/Unity-WebGL-SignalR#2]
The web clients for SignalR allow adding handlers for multiple different hub methods. For example:

```
connection.On<string>("MethodA", (message) =>
{
	// Do something
});

connection.On<string>("MethodB", (message) =>
{
	// Do something else
});
```

My understanding is there are limitations to what can be done with Unity WebGL.
From https://docs.unity3d.com/Manual/ScriptingRestrictions.html

> ### Calling managed methods from native code
> Managed methods that need to be marshaled to a C function pointer so that they can be called from native code have a few restructions on AOT platforms:
> 
> - The managed method must be a static method
> - The managed method must have the [MonoPInvokeCallback] attribute

Because of this, is it required to make an explicit method for each of the callbacks we would like to pass through JavaScript?
This would mean the above example would require one callback for each of the listeners as follows:
```
[MonoPInvokeCallback(typeof(Action<string>))]
public static void MethodACallback(string message)
{
	OnMethodAReceived(message);
}

[MonoPInvokeCallback(typeof(Action<string>))]
public static void MethodBCallback(string message)
{
	OnMethodBReceived(message);
}
```

Am I correct in my understanding here?

**Comment by evanlindsey** [comment]
@sstream17 Yes, with the current implementation, this would be necessary.

When using the editor, you can see that the handler matches your first example:
https://github.com/evanlindsey/Unity-WebGL-SignalR/blob/991ff277c82a798a954bba3a05a384c9976ea536/Unity/Assets/Plugins/SignalR/SignalRLib.cs#L27

When using the web, we are binding the name of the handler to the target callback. The function in the .jslib file then matches what is expected of a handler:
https://github.com/evanlindsey/Unity-WebGL-SignalR/blob/991ff277c82a798a954bba3a05a384c9976ea536/Unity/Assets/Plugins/SignalR/SignalRLib.cs#L71
https://github.com/evanlindsey/Unity-WebGL-SignalR/blob/991ff277c82a798a954bba3a05a384c9976ea536/Unity/Assets/Plugins/SignalR/SignalRLib.jslib#L15

It's possible this could be done more dynamically. I will tag the issue as an enhancement.

**Comment by sstream17** [comment]
@evanlindsey thanks for the response! I started looking into implementing a more dynamic solution which is similar to how the `On` implementation works in SignalR web clients. It uses a dictionary of handlers
https://github.com/dotnet/aspnetcore/blob/1a8f5a84fa8abfb7f72e2c0517f6a22ccc48e9b0/src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs#L75

This is where I ran into issues because of the `MonoPInvokeCallback` requirement.

**Comment by evanlindsey** [comment]
@sstream17 I was able to add support for multiple client handlers. It's included in the 0.0.2 release: 
https://github.com/evanlindsey/Unity-WebGL-SignalR/releases/tag/v0.0.2

Feel free to leave feedback on if this works for your use case. Closing out the issue for now!

**Comment by sstream17** [comment]
@evanlindsey sorry for the late response! The new version has been working great, thank you for adding it so quickly!
